gcloud/error_reporting/client.py

```diff
@@ -0,0 +1,145 @@
+                       :class:`NoneType`
+    :param credentials: The OAuth2 Credentials to use for the connection
+                        owned by this client. If not passed (and if no ``http``
+                        object is passed), falls back to the default inferred
+                        from the environment.
+
+    :type http: :class:`httplib2.Http` or class that defines ``request()``.
+    :param http: An optional HTTP object to make requests. If not passed, an
+                 ``http`` object is created that is bound to the
+                 ``credentials`` for the current object.
+
+    :type service: str
+    :param service: An identifier of the service, such as the name of the
+                    executable, job, or Google App Engine module name. This
+                    field is expected to have a low number of values that are
+                    relatively stable over time, as opposed to version,
+                    which can be changed whenever new code is deployed.
+
+
+    :type version: str
+    :param version: Represents the source code version that the developer
+                    provided, which could represent a version label or a Git
+                    SHA-1 hash, for example. If the developer did not provide
+                    a version, the value is set to default.
+
+    :raises: :class:`ValueError` if the project is neither passed in nor
+             set in the environment.
+    """
+
+    def __init__(self, project=None,
+                 credentials=None,
+                 http=None,
+                 service=None,
+                 version=None):
+        self.logging_client = gcloud.logging.client.Client(
+            project, credentials, http)
+        self.service = service
+        self.version = version
+
+    DEFAULT_SERVICE = 'python'
+    DEFAULT_VERSION = 'default'
+
+    def _get_default_service(self):
+        """Returns the service to use on method calls that don't specify an
+        override.
+
+          :rtype: string
+          :returns: The default service for error reporting calls
+        """
+        if self.service:
+            return self.service
+        else:
+            return self.DEFAULT_SERVICE
+
+    def _get_default_version(self):
+        """Returns the service to use on method calls that don't specify an
+        override.
+
+        :rtype: string
+        :returns: The default version for error reporting calls.
+        """
+        if self.version:
+            return self.version
+        else:
+            return self.DEFAULT_VERSION
+
+    def report_error(self, message="", service=None, version=None):
+        """ Reports the details of the latest exceptions to Stackdriver Error
+            Reporting.
+
+         https://cloud.google.com/error-reporting/docs/formatting-error-messages
+
+           :type message: str
+           :param message: An optional message to include with the exception
```

Comment:
@steren isn't `message` really optional to the caller of `Client.report_error` (as opposed to the back-end API it calls?)